modules/registration/functions/events.inc

```diff
@@ -137,3 +137,87 @@ function in_registration()
     }
 
 }
+
+
+function registration_report($event = null, $clause = null)
+{
+    if ($event === null) {
+        $event = \current_eventID();
+    }
+
+    return <<<SQL
+        SELECT
+            bt.`Name` BadgeType,
+            COUNT(DISTINCT rt.`AccountID`) AS TotalSold,
+            (
+            SELECT
+                COUNT(`AccountID`)
+            FROM
+                `Registrations`
+            WHERE
+                `BadgesPickedUp` > 0 AND BadgeTypeId = bt.BadgeTypeID
+        ) AS TotalPickedUp
+        FROM
+            `BadgeTypes` bt
+        INNER JOIN `Registrations` rt USING(BadgeTypeID)
+        WHERE
+            rt.EventID = '$event'
+        GROUP BY
+            bt.Name
+
+SQL;
+
+}
+
+
+function report_memberships_sold($event = null)
+{
+    if ($event === null) {
+        $event = \current_eventID();
+    }
+
+    return <<<SQL
+        SELECT
```

Comment:
Have I mentioned before how I **really** appreciate the way you write readable SQL queries :-D